odoo/tests/runner.py

```diff
@@ -18,10 +19,22 @@ def __init__(self):
         super().__init__()
         self.time_start = None
         self.queries_start = None
+        self._soft_fail = False
+        self.had_failure = False
 
     def __str__(self):
         return f'{len(self.failures)} failed, {len(self.errors)} error(s) of {self.testsRun} tests'
 
+    @contextlib.contextmanager
+    def soft_fail(self):
+        self.had_failure = False
```

Comment:
Is touching this really necessary given `startTest` will automatically update it?